File: app/src/main/java/poker/fivecardstats/activity/ScoreActivity.java
package poker.fivecardstats.activity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Spinner;
import android.widget.Toast;

import java.util.HashMap;
import java.util.List;

import poker.fivecardstats.R;
import poker.fivecardstats.adapter.ScoreAdapter;
import poker.fivecardstats.dao.ScoreDataSource;
import poker.fivecardstats.model.Score;

public class ScoreActivity extends AppCompatActivity {
    private ScoreDataSource mScoreDataSource;
    private FloatingActionButton fab;
    private List<Score> mScores;
    private ScoreAdapter mAdapter;
    private ListView listView;
    private long id;
    private Spinner spinner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_score);

        Intent user = getIntent();
        id = user.getLongExtra("id", 1L);

        init();
        initDB();
        listViewManager();
        fabManager();

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }

    private void listViewManager() {
        mScores = mScoreDataSource.getScores(id);
        mAdapter = new ScoreAdapter(this, mScores);
        listView.setAdapter(mAdapter);
    }

    private void init() {
        listView = (ListView) findViewById(R.id.scores_listview);
    }

    private void initDB() {
        mScoreDataSource = new ScoreDataSource(this);
        try {
            mScoreDataSource.open();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void fabManager() {
        fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                LayoutInflater layoutInflater = LayoutInflater.from(ScoreActivity.this);

                final View inflator = layoutInflater.inflate(R.layout.content_add_score, null);

                AlertDialog.Builder builder = new AlertDialog.Builder(ScoreActivity.this);
                builder.setTitle("Add new result");
                builder.setView(inflator);

                final HashMap<String, Integer> cardPoints = new HashMap<>();
                cardPoints.put("Par", 10);
                cardPoints.put("Två par", 20);
                cardPoints.put("Triss", 30);
                cardPoints.put("Stege", 40);
                cardPoints.put("Färg", 50);
                cardPoints.put("Kåk", 60);
                cardPoints.put("Fyrtal", 70);
                cardPoints.put("Royal street flush", 500);

                spinner = (Spinner) inflator.findViewById(R.id.card_points_spinner);
                ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(ScoreActivity.this,
                        R.array.card_score_array, android.R.layout.simple_spinner_dropdown_item);
                spinner.setAdapter(adapter);

                builder.setPositiveButton("Submit", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        String type = spinner.getSelectedItem().toString();
                        int score = cardPoints.get(type);
                        mScoreDataSource.create(id, type, score);

                        if(type.contains("Fyrtal")) {
                            mScoreDataSource.delete(id);
                        }
                        
                        updateList();
                    }
                }).show();
            }
        });
    }

    private void updateList() {
        mAdapter.clear();

        mScores = mScoreDataSource.getScores(id);
        mAdapter = new ScoreAdapter(this, mScores);
        listView.setAdapter(mAdapter);

        mAdapter.notifyDataSetChanged();
    }

    @Override
    public void onBackPressed() {
        finish();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/kotlin/poker/fivecardstats/activity/ScoreActivity.kt
package poker.fivecardstats.activity

import android.app.AlertDialog
import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Spinner
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.floatingactionbutton.FloatingActionButton
import poker.fivecardstats.R
import poker.fivecardstats.adapter.ScoreAdapter
import poker.fivecardstats.dao.ScoreDataSource
import poker.fivecardstats.model.Score

class ScoreActivity : AppCompatActivity() {
    private lateinit var mScoreDataSource: ScoreDataSource
    private lateinit var fab: FloatingActionButton
    private lateinit var mScores: List<Score>
    private lateinit var mAdapter: ScoreAdapter
    private lateinit var listView: ListView
    private lateinit var totalScoreTextView: TextView
    private var id: Long = 0
    private lateinit var spinner: Spinner

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_score)

        val user = intent
        id = user.getLongExtra("id", 1L)
        val name = user.getStringExtra("name")

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.title = name

        init()
        initDB()
        listViewManager()
        fabManager()
    }

    private fun listViewManager() {
        mScores = mScoreDataSource.getScores(id)
        mAdapter = ScoreAdapter(this, mScores)
        listView.adapter = mAdapter
        updateTotalScore()
    }

    private fun init() {
        listView = findViewById(R.id.scores_listview)
        totalScoreTextView = findViewById(R.id.total_score_textview)
    }

    private fun initDB() {
        mScoreDataSource = ScoreDataSource(this)
        try {
            mScoreDataSource.open()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun fabManager() {
        fab = findViewById(R.id.fab)
        fab.setOnClickListener { 
            val layoutInflater = LayoutInflater.from(this@ScoreActivity)

            val inflator = layoutInflater.inflate(R.layout.content_add_score, null)

            val builder = AlertDialog.Builder(this@ScoreActivity)
            builder.setTitle("Add new result")
            builder.setView(inflator)

            val cardPoints = hashMapOf(
                    "Par" to 10,
                    "Två par" to 20,
                    "Triss" to 30,
                    "Stege" to 40,
                    "Färg" to 50,
                    "Kåk" to 60,
                    "Fyrtal" to 70,
                    "Royal street flush" to 500
            )

            spinner = inflator.findViewById(R.id.card_points_spinner)
            val adapter = ArrayAdapter.createFromResource(this@ScoreActivity,
                    R.array.card_score_array, android.R.layout.simple_spinner_dropdown_item)
            spinner.adapter = adapter

            builder.setPositiveButton("Submit") { _, _ ->
                val type = spinner.selectedItem.toString()
                val score = cardPoints[type] ?: 0
                mScoreDataSource.create(id, type, score)

                if (type.contains("Fyrtal")) {
                    mScoreDataSource.delete(id)
                }

                updateList()
            }.show()
        }
    }

    private fun updateList() {
        mScores = mScoreDataSource.getScores(id)
        mAdapter.clear()
        mAdapter.addAll(mScores)
        updateTotalScore()
    }

    private fun updateTotalScore() {
        val totalScore = mScores.sumOf { it.score }
        totalScoreTextView.text = "Total: ${totalScore}p"
    }

    override fun onBackPressed() {
        finish()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        if (id == R.id.action_settings) {
            return false
        }
        return super.onOptionsItemSelected(item)
    }
}